Cargo.lock

```diff
@@ -1572,8 +1558,8 @@ dependencies = [
  "http",
  "indexmap",
  "slab",
- "tokio 1.1.1",
```

Comment:
By design, it's really easy to introduce duplicate dependencies with rust. Sometimes it's helpful, and I'm glad it's possible in general, but just as often it's  problematic. 

In the best case, it's just some bloat, in the bad case you'll get compiler errors due to different version of structs/traits trying to interact and in the worst case, like here, things will just work and you'll get a runtime error due to "not being on the tokio runtime" when what they really mean is "you are on a different version of the tokio runtime"

Probably we could do somethign fancy and automatic to be more on alert when we're adding dupe dependencies, maybe with `cargo tree -d` or something, but my current approach is to be on the lookout for lines like this whenever Cargo.lock changes - if a version string is newly added to a dependency, it's because we've introduced a duplicate and now the crate must be disambiguated. 

Sorry if you already knew all of this.

Reply:
I knew most of this, but thanks for spelling it out so clearly! If we ever got to a point with no repeat deps, we might be able to write a query using `cargo tree`, and then add a CI script to enforce no regressions. But in the meantime, watching the diff on Cargo.lock seems reasonable